lib/js/src/manager/lifecycle/LifecycleManager.js

```diff
@@ -30,8 +30,291 @@
+                    this._responseListeners.set(rpcMessage.getCorrelationId(), rpcMessage.getOnRPCResponsePromise());
+                }
+            }
+            this._sdlSession.sendRpc(rpcMessage);
+        }
+    }
+
+
+
+    /**
+     * @return {RegisterAppInterfaceResponse}
+     */
+    getRegisterAppInterfaceResponse() {
+        return this._registerAppInterfaceResponse;
+    }
+
+
+
+    /**
+    * @return {RegisterAppInterface}
+    */
+    _createRegisterAppInterface() {
+        const registerAppInterface = new RegisterAppInterface();
+        registerAppInterface.setSdlMsgVersion(new SdlMsgVersion().setMajorVersion(LifecycleManager.MAX_RPC_VERSION.getMajor()).setMinorVersion(LifecycleManager.MAX_RPC_VERSION.getMinor()).setPatchVersion(LifecycleManager.MAX_RPC_VERSION.getPatch()))
+            .setAppName(this._appConfig.getAppName())
+            .setFullAppId(this._appConfig.getAppId())
+            .setNgnMediaScreenAppName(this._appConfig.getShortAppName())
+            .setAppHmiType(this._appConfig.getAppTypes())
+            .setLanguageDesired(this._appConfig.getLanguageDesired())
+            .setHmiDisplayLanguageDesired(this._appConfig.getHmiDisplayLanguageDesired())
+            .setIsMediaApplication(this._appConfig.isMediaApp())
+            .setDayColorScheme(this._appConfig.getDayColorScheme())
+            .setNightColorScheme(this._appConfig.getNightColorScheme())
+            .setCorrelationId(LifecycleManager.REGISTER_APP_INTERFACE_CORRELATION_ID);
+
+        //TODO Add all possible items
+
+        return registerAppInterface;
+    }
+
+
+    /* *******************************************************************************************************
+     ********************************** INTERNAL - RPC LISTENERS !! START !! *********************************
+     *********************************************************************************************************/
+
+    _setupInternalRpcListeners() {
+        this._rpcListener = new RpcListener().setOnRpcMessage(rpcMessage => {
+            const functionID = FunctionID.valueForString(rpcMessage.getFunctionName()); // this is the number value
+
+
+            switch (functionID) {
+                case FunctionID.RegisterAppInterface:
+                    this._processRaiResponse(rpcMessage);
+                    break;
+                case FunctionID.OnHMIStatus:
+                    // send a single onProxyConnected, when we go from a null HMI level to a defined HMI level
+                    const shouldInit = rpcMessage.getHMILevel() !== null 
+                        && rpcMessage.getHMILevel() !== undefined
+                        && this._currentHMIStatus === null;
+
+                    this._currentHMIStatus = rpcMessage.getHMILevel();
+
+                    if (this._lifecycleListener !== null && this._lifecycleListener !== undefined && shouldInit) {
+                        this._lifecycleListener.onProxyConnected(this);
```

Comment:
Can the other lifecycle listener methods be used in this file? There's also onProxyClosed, onServiceStarted, onServiceEnded, onError.

Reply:
I think when we create the `SdlManager` class more of that will become clear. We should take another look at that time to see if we need those methods, or need other methods.